src/AdvancedMH.jl

```diff
@@ -53,17 +53,21 @@ end
 
 const DensityModelOrLogDensityModel = Union{<:DensityModel,<:AbstractMCMC.LogDensityModel}
 
-# Create a very basic Transition type, only stores the
-# parameter draws and the log probability of the draw.
-struct Transition{T,L<:Real} <: AbstractTransition
+# Create a very basic Transition type, stores the
+# parameter draws, the log probability of the draw,
+# and the draw information until this point
+struct Transition{T,L<:Real,M<:Bool,D<:Int} <: AbstractTransition
```

Comment:
These type parameters are not needed actually, only `Bool` and `Int` can subtype `Bool` and `Int`, respectively.
```suggestion
struct Transition{T,L<:Real} <: AbstractTransition
```

Reply:
I think all of the changes I've made to `Transition` will have to be mirrored in Turing. Otherwise, I think it will break all over the place. I found several places where `Transitions` are being constructed.